portsyncd/portsyncd.cpp

```diff
@@ -199,7 +199,7 @@ void handlePortConfigFile(ProducerStateTable &p, string file)
         throw "Port configuration file not found!";
     }
 
-    list<string> header = {"name", "lanes", "alias", "speed"};
+    list<string> header = {"name", "lanes", "alias", "speed", "autoneg", "fec"};
```

Comment:
in the master branch, fec is coming from config. If we are going to support autoneg in master, it should also come from config db.

Reply:
Isn't there a problem with config DB for initial port states - ports will come up as default and then potentially "flap" when this configuration from DB is applied? I thought the portconfig.ini is first played into the SAI library and then modified when the config DB is evaluated and it notices a change. I may be wrong on this.

Reply:
ultimately, all the configurations should come from one place. we plan to consolidate all configurations into config db. 

I think since we are using create port API with the attributes coming from config db, there should be no flap.

Reply:
So, we can remove the portconfig.ini file and get the breakout, speeds and hostinterface name, index associations from the configuration DB. The problem we had was that at boot up - we come up with default configurations - if the default matches the portconfig.ini no changes are observed. If changes come later it causes the remote nodes to see another flap. If we can eliminate this it will be great. Currently, you can see that some of the device configurations use breakout, fec and AN in the default config files. If you want us to remove fec and autoneg in  the 201712 branch, we can do it. But the devices will need to apply a patch to build for their platforms. Some of them have fixed breakout configuration and default to fec and AN settings (sonic-buildimage pull request). Let me know if you want me to revert the changes and I will do so.

Reply:
sorry for the confusion, I am ok with this on 201712 branch. For master branch, we need to take a different approach.